### PR TITLE
[python] Added draw_line function in kandinsky python module

### PR DESCRIPTION
**Pull request by RedGl0w**
## Patch

https://github.com/numworks/epsilon/issues/1291

## Exemple
image
```py
from kandinsky import *
fill_rect(0,0,300,300,color(0,0,0))
draw_line(0,0,300,300,color(255,255,255))
```

## TODO

- [x] Add in the toolbox the function
- [ ] Find shorter translations

### CODE REVIEW

File: apps/code/catalog.en.i18n [hunk 1]
```diff
@@ -29,6 +29,7 @@ PythonCosh = "Hyperbolic cosine"
 PythonDegrees = "Convert x from radians to degrees"
 PythonDivMod = "Quotient and remainder"
 PythonDrawString = "Display a text from pixel (x,y)"
+PythonDrawLine = "Makes a line from (x1, y1) to (x2, y2)"
```
Comment (Tim-ats-d) [inline apps/code/catalog.en.i18n]:
```suggestion
PythonDrawLine = "Draw a line from (x1, y1) to (x2, y2)"
```

File: apps/code/catalog.de.i18n [hunk 2]
```diff
@@ -29,6 +29,7 @@ PythonCosh = "Hyperbolic cosine"
 PythonDegrees = "Convert x from radians to degrees"
 PythonDivMod = "Quotient and remainder"
 PythonDrawString = "Display a text from pixel (x,y)"
+PythonDrawLine = "Erstellt eine Linie von (x1, y1) nach (x2, y2)"
```
Comment (Tim-ats-d) [inline apps/code/catalog.de.i18n]:
```suggestion
PythonDrawLine = "Zeichne eine Linie von (x1, y1) nach (x2, y2)"
```

File: apps/code/catalog.fr.i18n [hunk 3]
```diff
@@ -29,6 +29,7 @@ PythonCosh = "Cosinus hyperbolique"
 PythonDegrees = "Conversion de radians en degrés"
 PythonDivMod = "Quotient et reste"
 PythonDrawString = "Affiche un texte au pixel (x,y)"
+PythonDrawLine = "Fait une ligne de (x1,y1) à (x2,y2)"
```
Comment (Tim-ats-d) [inline apps/code/catalog.fr.i18n]:
```suggestion
PythonDrawLine = "Trace une ligne de (x1,y1) à (x2,y2)"
```

File: apps/code/catalog.es.i18n [hunk 4]
```diff
@@ -29,6 +29,7 @@ PythonCosh = "Hyperbolic cosine"
 PythonDegrees = "Convert x from radians to degrees"
 PythonDivMod = "Quotient and remainder"
 PythonDrawString = "Display a text from pixel (x,y)"
+PythonDrawLine = "Hace una línea de (x1, y1) a (x2, y2)"
```
Comment (Tim-ats-d) [inline apps/code/catalog.es.i18n]:
```suggestion
PythonDrawLine = "Dibujar una línea de (x1, y1) a (x2, y2)"
```

File: apps/code/catalog.pt.i18n [hunk 5]
```diff
@@ -29,6 +29,7 @@ PythonCosh = "Hyperbolic cosine"
 PythonDegrees = "Convert x from radians to degrees"
 PythonDivMod = "Quotient and remainder"
 PythonDrawString = "Display a text from pixel (x,y)"
+PythonDrawLine = "Faz uma linha de (x1, y1) a (x2, y2)"
```
Comment (Tim-ats-d) [inline apps/code/catalog.pt.i18n]:
```suggestion
PythonDrawLine = "Desenhe uma linha de (x1, y1) a (x2, y2)"
```

File: apps/code/catalog.es.i18n [hunk 6]
```diff
@@ -29,6 +29,7 @@ PythonCosh = "Hyperbolic cosine"
 PythonDegrees = "Convert x from radians to degrees"
 PythonDivMod = "Quotient and remainder"
 PythonDrawString = "Display a text from pixel (x,y)"
+PythonDrawLine = "Dibujar una línea de (x1, y1) a (x2, y2)"
```
Comment (Tim-ats-d) [inline apps/code/catalog.es.i18n]:
```suggestion
PythonDrawLine = "Dibujar una línea de (x1,y1) a (x2,y2)"
```

File: apps/code/catalog.pt.i18n [hunk 7]
```diff
@@ -29,6 +29,7 @@ PythonCosh = "Hyperbolic cosine"
 PythonDegrees = "Convert x from radians to degrees"
 PythonDivMod = "Quotient and remainder"
 PythonDrawString = "Display a text from pixel (x,y)"
+PythonDrawLine = "Desenhe uma linha de (x1, y1) a (x2, y2)"
```
Comment (Tim-ats-d) [inline apps/code/catalog.pt.i18n]:
```suggestion
PythonDrawLine = "Desenhe uma linha de (x1,y1) a (x2,y2)"
```

File: apps/code/catalog.de.i18n [hunk 8]
```diff
@@ -29,6 +29,7 @@ PythonCosh = "Hyperbolic cosine"
 PythonDegrees = "Convert x from radians to degrees"
 PythonDivMod = "Quotient and remainder"
 PythonDrawString = "Display a text from pixel (x,y)"
+PythonDrawLine = "Zeichne eine Linie von (x1,y1) nach (x2,y2)"
```
Comment (Tim-ats-d) [inline apps/code/catalog.de.i18n]:
```suggestion
PythonDrawLine = "Zeichne eine Linie von x1 y1 nach x2 y2"
```

File: apps/code/catalog.en.i18n [hunk 9]
```diff
@@ -29,6 +29,7 @@ PythonCosh = "Hyperbolic cosine"
 PythonDegrees = "Convert x from radians to degrees"
 PythonDivMod = "Quotient and remainder"
 PythonDrawString = "Display a text from pixel (x,y)"
+PythonDrawLine = "Draw a line from (x1,y1) to (x2,y2)"
```
Comment (Tim-ats-d) [inline apps/code/catalog.en.i18n]:
```suggestion
PythonDrawLine = "Draw a line from x1 y1 to x2 y2"
```

File: apps/code/catalog.es.i18n [hunk 10]
```diff
@@ -29,6 +29,7 @@ PythonCosh = "Hyperbolic cosine"
 PythonDegrees = "Convert x from radians to degrees"
 PythonDivMod = "Quotient and remainder"
 PythonDrawString = "Display a text from pixel (x,y)"
+PythonDrawLine = "Dibujar una línea de (x1,y1) a (x2,y2)"
```
Comment (Tim-ats-d) [inline apps/code/catalog.es.i18n]:
```suggestion
PythonDrawLine = "Dibujar una línea de x1 y1 a x2 y2"
```

File: apps/code/catalog.fr.i18n [hunk 11]
```diff
@@ -29,6 +29,7 @@ PythonCosh = "Cosinus hyperbolique"
 PythonDegrees = "Conversion de radians en degrés"
 PythonDivMod = "Quotient et reste"
 PythonDrawString = "Affiche un texte au pixel (x,y)"
+PythonDrawLine = "Trace une ligne de (x1,y1) à (x2,y2)"
```
Comment (Tim-ats-d) [inline apps/code/catalog.fr.i18n]:
```suggestion
PythonDrawLine = "Trace une ligne de x1 y1 à x2 y2"
```

File: apps/code/catalog.pt.i18n [hunk 12]
```diff
@@ -29,6 +29,7 @@ PythonCosh = "Hyperbolic cosine"
 PythonDegrees = "Convert x from radians to degrees"
 PythonDivMod = "Quotient and remainder"
 PythonDrawString = "Display a text from pixel (x,y)"
+PythonDrawLine = "Desenhe uma linha de (x1,y1) a (x2,y2)"
```
Comment (Tim-ats-d) [inline apps/code/catalog.pt.i18n]:
```suggestion
PythonDrawLine = "Desenhe uma linha de x1 y1 a x2 y2"
```

**Comment by artaxxx** [comment]
Thanks @RedGl0w ! It could be useful :)

I see some problems here:
* on your screenshot, it doesn't look like a line, know why?
* the following script doesn't work:

```python
from kandinsky import *
draw_line(0,0,300,300,color(255,0,0))
```

The line is drawn but instantly disappears. It should be persistent.

**Comment by RedGl0w** [comment]
>     * on your screenshot, it doesn't look like a line, know why?

There isn't anti aliasing, maybe that's what why it doesn't look like a line ?

> The line is drawn but instantly disappears. It should be persistent.

I'll try to understand, and reopen a PR quickly

**Comment by artaxxx** [comment]
This is not about anti-aliasing: it's OK on the device but seems odd on your screenshot.

**Comment by RedGl0w** [comment]
Oh yes you're right
image
I'll try to understand in Kandinsky why it reacts like that :thinking:

**Pull request closed without merging**